packages/theme/src/cli/utilities/theme-environment/local-assets.ts

```diff
@@ -94,35 +94,143 @@ function isCompiledAssetRequest(event: H3Event) {
 function handleCompiledAssetRequest(event: H3Event, ctx: DevServerContext) {
   const assetPath = parseServerEvent(event).pathname.split('/').at(-1)
 
-  if (assetPath === 'styles.css') {
-    const allLiquidFiles = [...ctx.localThemeFileSystem.files.entries()].filter(([key]) => key.endsWith('.liquid'))
+  switch (assetPath) {
+    case 'styles.css':
+      return handleStylesCss(ctx, event)
+    case 'block-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'block')
+    case 'snippet-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'snippet')
+    case 'section-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'section')
+    default:
+      return handleError(event)
+  }
+}
 
-    const getLiquidFilesFromDir = (dirPrefix: string) => {
-      return allLiquidFiles
-        .filter(([key]) => key.startsWith(`${dirPrefix}/`))
-        .sort(([key1], [key2]) => key1.localeCompare(key2))
-    }
+const compiledStylesheets = new Map<string, {checksum: string; content: string}>()
+const compiledJavaScripts = new Map<string, {checksum: string; content: string}>()
+
+function handleStylesCss(ctx: DevServerContext, event: H3Event) {
+  const sectionFiles = getLiquidFilesByKind(ctx, 'section')
+  const blockFiles = getLiquidFilesByKind(ctx, 'block')
+  const snippetFiles = getLiquidFilesByKind(ctx, 'snippet')
+
+  const stylesheets = ['/*** GENERATED LOCALLY ***/\n']
 
-    const sectionFiles = getLiquidFilesFromDir('sections')
-    const blockFiles = getLiquidFilesFromDir('blocks')
-    const snippetFiles = getLiquidFilesFromDir('snippets')
+  for (const [, file] of [...sectionFiles, ...blockFiles, ...snippetFiles]) {
+    const stylesheet = getCompiledStylesheet(file)
+    if (stylesheet) stylesheets.push(stylesheet)
+  }
+
+  const stylesheet = stylesheets.join('\n')
+
+  return serveStatic(event, {
+    getContents: () => stylesheet,
+    getMeta: () => ({type: 'text/css', size: stylesheet.length, mtime: new Date()}),
+  })
+}
 
-    const stylesheets = ['/*** GENERATED LOCALLY ***/\n']
-    for (const [, file] of [...sectionFiles, ...blockFiles, ...snippetFiles]) {
-      const stylesheet = getCompiledStylesheet(file)
-      if (stylesheet) stylesheets.push(stylesheet)
+function handleBlockScriptsJs(ctx: DevServerContext, event: H3Event, kind: 'block' | 'snippet' | 'section') {
+  const blockFiles = getLiquidFilesByKind(ctx, kind)
```

Comment:
Is `blockFiles` naming correct here? It can be anything, right? 🤔

Reply:
Great catch! Fixed :)